lib/client.js

```diff
@@ -53,8 +54,12 @@ function Client(config = {}) {
   if (config.version) {
     this.version = config.version;
   }
-
-  this.connectionSettings = cloneDeep(config.connection || {});
+  // if the type of connection settings is a function, pass it to the next layer
+  if (typeof config.connection === 'function') {
```

Comment:
I'd make this whole thing more encapsulated:
```
this.connectionSettingsProvider = new ConnectionSettingsProvider(config.connection)
```
And move the isFunction branching inside provider itself.
You don't even need dependency on the client, I think.

Reply:
yes, you are right. I've removed connection handling in client.js and moved it to connectionProvider.js